### PR TITLE
Fix edgegateway retrieval from VDC group

### PR DESCRIPTION
**Pull request by dataclouder**
Address Issue #1127

Fix incorrect behavior of `vcd_resource_list`, which can retrieve edge gateways belonging to a VDC, but not belonging to a VDC group.

### CODE REVIEW

File: vcd/datasource_vcd_resource_list.go [hunk 1]
```diff
@@ -590,7 +590,18 @@ func orgNetworkListV2(d *schema.ResourceData, meta interface{}) (list []string,
...
 
-	org, vdc, err := client.GetOrgAndVdc(d.Get("org").(string), d.Get("vdc").(string))
+	vdcName := d.Get("parent").(string)
+	if vdcName == "" {
+		vdcName = d.Get("vdc").(string)
+	}
+	if vdcName == "" {
+		vdcName = client.Vdc
+	}
+
```
Comment (lvirbalas) [inline vcd/datasource_vcd_resource_list.go:600]: Nit. Extra empty line (which actually makes these ifs and their grouping look stranger than they are).

File: vcd/datasource_vcd_resource_list.go [hunk 2]
```diff
@@ -590,7 +590,18 @@ func orgNetworkListV2(d *schema.ResourceData, meta interface{}) (list []string,
...
+	if vdcName == "" {
+		return nil, fmt.Errorf("VDC name not given either as 'vdc' or 'parent' field")
```
Comment (lvirbalas) [inline vcd/datasource_vcd_resource_list.go:602]: Maybe you're using these "three ifs to choose at least one of available VDC name fields" in other places too, and it's worth having it in a function? (In an effort to make this code part easier on the eyes).

Comment (dataclouder) [reply vcd/datasource_vcd_resource_list.go:617]: Function added. Some places use it now

Comment (lvirbalas) [reply vcd/datasource_vcd_resource_list.go:617]: Neat! Thanks!

File: .changes/v3.11.0/1129-bug-fixes.md [hunk 3]
```diff
@@ -0,0 +1 @@
+* Fix Issue [#1127](https://github.com/vmware/terraform-provider-vcd/issues/1127) (Incorrect behavior of vcd_resource_list, which can retrieve edge gateways belonging to a VDC, but not belonging to a VDC group) [GH-1129]
```
Comment (lvirbalas) [inline .changes/v3.11.0/1129-bug-fixes.md:1]: Nit. In this context, capitalizing keywords would be helpful:

```suggestion
* Fix Issue [#1127](https://github.com/vmware/terraform-provider-vcd/issues/1127) (Incorrect behavior of vcd_resource_list, which can retrieve Edge Gateways belonging to a VDC, but not belonging to a VDC Group) [GH-1129]
```

**Review by lvirbalas — APPROVED** [review]
One more small ask in-line, and looks great!

File: .changes/v3.11.0/1129-bug-fixes.md (hunk 3 above)
Comment (dataclouder) [reply .changes/v3.11.0/1129-bug-fixes.md:1]: done

**Review by Didainius — APPROVED** [review]

**Review by adezxc — APPROVED** [review]

**Review by adambarreiro — APPROVED** [review]
LGTM

**Pull request merged**